test/configCases/rebuild/finishModules/webpack.config.js

```diff
@@ -0,0 +1,43 @@
+const { resolve, join } = require("path");
+
+/**
+ * @this {import("../../../../").Compiler} the compiler
+ */
+var testPlugin = function () {
+	this.hooks.compilation.tap("TestPlugin", compilation => {
+		compilation.hooks.finishModules.tapAsync("TestPlugin", function (
+			modules,
+			callback
+		) {
+			const src = resolve(join(__dirname, "other-file.js"));
+
+			/**
+			 *
+			 * @param {any} m test
+			 * @returns {boolean} test
+			 */
+			function matcher(m) {
+				return m.resource && m.resource === src;
+			}
+
+			const module = Array.from(compilation.modules).find(matcher);
+			/** @type {any} */
+			const inputFileSystem = compilation.inputFileSystem;
+			const cachedFileInput = inputFileSystem._readFileBackend._data.get(src);
+			cachedFileInput.result = `module.exports = { foo: { foo: 'bar' }, doThings: () => { }}`;
```

Comment:
Not a fan of the way in which this has to break type-safety, and dive into internal\private methods to achieve this test.

But by doing this I can test that the module has been rebuilt.

As mentioned in the description, one idea was to change the contents of the file in a second run through the use of a loader, but I'm not super comfortable with my ability to write a custom loader.